src/plot_api/plot_api.js

```diff
@@ -3248,9 +3252,6 @@ function makePlotFramework(gd) {
     var gd3 = d3.select(gd);
     var fullLayout = gd._fullLayout;
 
-    // Check if gd has a specified height
-    fullLayout._hasZeroHeight = gd.clientHeight === 0;
```

Comment:
The reason I removed _hasZeroHeight from fullLayout and put it into context is because it would get cleared by a call to `Plotly.newPlot`. However, because this call does not remove the plot from the container, `gd.clientHeight` would be non-zero. This is not because the container has a specified width/height in CSS but simply because it expanded for the plot. As soon as the plot is removed, it goes back to being zero size. Not good!

Reply:
Ok. That looks good to me. Stashing those things `_context` is "great", but I can't think of a better place, so :ok_hand: 

The problems brought up in https://github.com/plotly/plotly.js/pull/3056#issuecomment-428305909 seemed to now be resolved.

I'll test this thing on stage after lunch.